### ISSUE TITLE
Error updating 5.1.2 to 6 samba dependency.

### ISSUE THREAD

**Issue by neovs** [zentyal/zentyal#1843]
When upgrading, I get an error from the package (- configure):
 problems of dependencies - we leave not configured.
Attempting to reinstall gives the same error.

sudo dpkg --configure -a
The samba package is configured (2: 4.7.6 + dfsg ~ ubuntu-0ubuntu2.2) ...
initctl: Unable to connect to Upstart: Failed to connect to socket / com / ubuntu / upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel (s) for script `lxdm '
insserv: Default-Stop undefined, assuming empty stop runlevel (s) for script `lxdm '
initctl: Unable to connect to Upstart: Failed to connect to socket / com / ubuntu / upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel (s) for script `lxdm '
insserv: Default-Stop undefined, assuming empty stop runlevel (s) for script `lxdm '
initctl: Unable to connect to Upstart: Failed to connect to socket / com / ubuntu / upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel (s) for script `lxdm '
insserv: Default-Stop undefined, assuming empty stop runlevel (s) for script `lxdm '
Job for samba-ad-dc.service failed because a timeout was exceeded.
See "systemctl status samba-ad-dc.service" and "journalctl-xe" for details.
invoke-rc.d: initscript samba-ad-dc, action "restart" failed.
● samba-ad-dc.service - Samba AD Daemon
   Loaded: loaded (/lib/systemd/system/samba-ad-dc.service; enabled; vendor preset: enabled)
   Active: failed (Result: timeout) since Sun 2018-11-18 22:17:44 MSK; 8ms ago
     Docs: man: samba (8)
           man: samba (7)
           man: smb.conf (5)
  Process: 7431 ExecStart = / usr / sbin / samba --foreground --no-process-group $ SAMBAOPTIONS (code = exited, status = 127)
 Main PID: 7431 (code = exited, status = 127)

Nov 18 22:17:44 dc1 samba [7444]: Exiting pid 7444 on SIGTERM
Nov 18 22:17:44 dc1 samba [7438]: [2018/11/18 22: 17: 44.980102, 0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
Nov 18 22:17:44 dc1 samba [7441]: [2018/11/18 22: 17: 44.980660, 0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
Nov 18 22:17:44 dc1 samba [7441]: Exiting pid 7441 on SIGTERM
Nov 18 22:17:44 dc1 systemd [1]: samba-ad-dc.service: Main process exited, code = exited, status = 127 / n / a
Nov 18 22:17:44 dc1 samba [7440]: [2018/11/18 22: 17: 44.978740, 0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
Nov 18 22:17:44 dc1 samba [7440]: Exiting pid 7440 on SIGTERM
Nov 18 22:17:44 dc1 samba [7438]: Exiting pid 7438 on SIGTERM
Nov 18 22:17:44 dc1 systemd [1]: samba-ad-dc.service: Failed with result 'timeout'.
Nov 18 22:17:44 dc1 systemd [1]: Failed to start Samba AD Daemon.
dpkg: error while processing the samba package (--configure):
 installed samba package post-installation script subprocess returned error exit status 1
dpkg: package dependencies do not allow configuring zentyal-samba package:
 zentyal-samba depends on samba, however:
  The samba package is not yet configured.

dpkg: error while processing the zentyal-samba package (--configure):
 dependency problems - leave not configured
Triggers for libc-bin (2.27-3ubuntu1) are processed ...
Errors occurred while processing the following packages:
 samba
 zentyal-samba

**Comment by neovs** [comment]
Updated zentyla to 5.13 error remained.


root @ dc1: / home / administrator # sudo dpkg --configure -a
The samba package is configured (2: 4.7.6 + dfsg ~ ubuntu-0ubuntu2.4) ...
initctl: Unable to connect to Upstart: Failed to connect to socket / com / ubuntu / u pstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel (s) for script `l xdm '
insserv: Default-Stop undefined, assuming empty stop runlevel (s) for script `l xdm '
initctl: Unable to connect to Upstart: Failed to connect to socket / com / ubuntu / u pstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel (s) for script `l xdm '
insserv: Default-Stop undefined, assuming empty stop runlevel (s) for script `l xdm '
initctl: Unable to connect to Upstart: Failed to connect to socket / com / ubuntu / u pstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel (s) for script `l xdm '
insserv: Default-Stop undefined, assuming empty stop runlevel (s) for script `l xdm '
Job for samba-ad-dc.service failed because a timeout was exceeded.
See "systemctl status samba-ad-dc.service" and "journalctl-xe" for details.
invoke-rc.d: initscript samba-ad-dc, action "restart" failed.
● samba-ad-dc.service - Samba AD Daemon
   Loaded: loaded (/lib/systemd/system/samba-ad-dc.service; enabled; vendor preset: enabled)
   Active: failed (Result: timeout) since Wed 2018-11-21 13:33:02 MSK; 11ms ago
     Docs: man: samba (8)
           man: samba (7)
           man: smb.conf (5)
  Process: 2402 ExecStart = / usr / sbin / samba --foreground --no-process-group $ SAMBAOPTIONS (code = exited, status = 127)
 Main PID: 2402 (code = exited, status = 127)

Nov 21 13:33:02 dc1 samba [2410]: Exiting pid 2410 on SIGTERM
Nov 21 13:33:02 dc1 samba [2404]: [2018/11/21 13: 33: 02.736628, 0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
Nov 21 13:33:02 dc1 samba [2404]: Exiting pid 2404 on SIGTERM
Nov 21 13:33:02 dc1 samba [2414]: [2018/11/21 13: 33: 02.738312, 0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
Nov 21 13:33:02 dc1 samba [2414]: Exiting pid 2414 on SIGTERM
Nov 21 13:33:02 dc1 samba [2406]: [2018/11/21 13: 33: 02.742106, 0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
Nov 21 13:33:02 dc1 samba [2406]: Exiting pid 2406 on SIGTERM
Nov 21 13:33:02 dc1 systemd [1]: samba-ad-dc.service: Main process exited, code = exited, status = 127 / n / a
Nov 21 13:33:02 dc1 systemd [1]: samba-ad-dc.service: Failed with result 'timeout'.
Nov 21 13:33:02 dc1 systemd [1]: Failed to start Samba AD Daemon.
dpkg: error while processing the samba package (--configure):
 installed samba package post-installation script subprocess returned error exit status 1
dpkg: package dependencies do not allow configuring zentyal-samba package:
 zentyal-samba depends on samba, however:
  The samba package is not yet configured.

dpkg: error while processing the zentyal-samba package (--configure):
 dependency problems - leave not configured
Triggers for libc-bin (2.27-3ubuntu1) are processed ...
Errors occurred while processing the following packages:
 samba
 zentyal-samba

**Comment by loxK** [comment]
I have the save issue should have looked at bugs before upgrading ...

**Comment by loxK** [comment]
I disabled the smbd service and could go sudo dpkg --configure -a

**Comment by gadgetusaf** [comment]
I am also getting an error
root@domain1:/home/burton# dpkg --configure -a
Setting up samba (2:4.7.6+dfsg~ubuntu-0ubuntu2.5) ...
initctl: Unable to connect to Upstart: Failed to connect to socket /com/ubuntu/upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel(s) for script `lxdm'
insserv: Default-Stop  undefined, assuming empty stop  runlevel(s) for script `lxdm'
initctl: Unable to connect to Upstart: Failed to connect to socket /com/ubuntu/upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel(s) for script `lxdm'
insserv: Default-Stop  undefined, assuming empty stop  runlevel(s) for script `lxdm'
initctl: Unable to connect to Upstart: Failed to connect to socket /com/ubuntu/upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel(s) for script `lxdm'
insserv: Default-Stop  undefined, assuming empty stop  runlevel(s) for script `lxdm'

purged samba and reinstalled with same error

**Comment by phaidros7** [comment]
same problem here. 

root@venice:~# dpkg --configure -a
Setting up zentyal-samba (6.0.1) ...
dienst  gibt es schon.dpkg: error processing package zentyal-samba (--configure):
 installed zentyal-samba package post-installation script subprocess returned error exit status 25
Errors were encountered while processing:
 zentyal-samba

somehow the service samba or alike is already there and dpkg gets confused. right now I cannot start samba (zentyal samba module, and therefor no pdc and file shares are available) .. 

any insights how to convince dpkg to cleanly install the update for zentyal-samba?

**Comment by phaidros7** [comment]
/var/lib/dpkg/info/zentyal-samba.postinst is the culprit, tho I have no idea about the reasons of this postinst script

**Comment by gadgetusaf** [comment]
ommand output: . 
Exit value: 1
2018/12/10 08:58:57 ERROR> Service.pm:971 EBox::Module::Service::restartService - root command systemctl start 'samba-ad-dc' failed. 
Error output: Job for samba-ad-dc.service failed because a timeout was exceeded.
 See "systemctl status samba-ad-dc.service" and "journalctl -xe" for details.

Command output: . 
Exit value: 1 at root command systemctl start 'samba-ad-dc' failed. 
Error output: Job for samba-ad-dc.service failed because a timeout was exceeded.
 See "systemctl status samba-ad-dc.service" and "journalctl -xe" for details.

Command output: . 
Exit value: 1 at /usr/share/perl5/EBox/Module/Service.pm line 971
EBox::Module::Service::restartService('EBox::Samba=HASH(0x558a13aabc68)', 'restartModules', 1) called at /usr/share/perl5/EBox/Util/Init.pm line 121
eval {...} at /usr/share/perl5/EBox/Util/Init.pm line 119
EBox::Util::Init::moduleAction('samba', 'restartService', 'start') called at /usr/share/perl5/EBox/Util/Init.pm line 87
EBox::Util::Init::start at /usr/bin/zs line 35
main::main at /usr/bin/zs line 82
2018/12/10 08:58:57 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: webadmin
2018/12/10 08:58:59 WARN> WebAdmin.pm:432 EBox::WebAdmin::_CAs - Ignoring CA /var/lib/zentyal/conf/remoteservices/subscription/Domain1/cacert.pem: cannot read the file or not is a regular file
2018/12/10 08:59:00 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: webadmin
2018/12/10 08:59:00 WARN> WebAdmin.pm:432 EBox::WebAdmin::_CAs - Ignoring CA /var/lib/zentyal/conf/remoteservices/subscription/Domain1/cacert.pem: cannot read the file or not is a regular file
2018/12/10 08:59:00 INFO> Init.pm:90 EBox::Util::Init::start - Start modules finished
2018/12/10 08:59:38 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: samba
2018/12/10 09:01:07 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: webadmin
2018/12/10 09:01:10 WARN> WebAdmin.pm:432 EBox::WebAdmin::_CAs - Ignoring CA /var/lib/zentyal/conf/remoteservices/subscription/Domain1/cacert.pem: cannot read the file or not is a regular file
2018/12/10 09:01:42 INFO> GlobalImpl.pm:616 EBox::GlobalImpl::saveAllModules - Saving config and restarting services: firewall dns dhcp samba logs
2018/12/10 09:01:43 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: firewall
2018/12/10 09:01:43 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: dns
2018/12/10 09:01:47 INFO> DNS.pm:91 EBox::DNS::appArmorProfiles - Setting DNS apparmor profile
2018/12/10 09:01:51 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: dhcp
2018/12/10 09:01:53 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: samba
2018/12/10 09:03:25 ERROR> Sudo.pm:240 EBox::Sudo::_rootError - root command systemctl start 'samba-ad-dc' failed. 
Error output: Job for samba-ad-dc.service failed because a timeout was exceeded.
 See "systemctl status samba-ad-dc.service" and "journalctl -xe" for details.

Command output: . 
Exit value: 1 at root command systemctl start 'samba-ad-dc' failed. 
Error output: Job for samba-ad-dc.service failed because a timeout was exceeded.
 See "systemctl status samba-ad-dc.service" and "journalctl -xe" for details.

Command output: . 
Exit value: 1 at /usr/share/perl5/EBox/Sudo.pm line 240
EBox::Sudo::_rootError('/usr/bin/sudo -p sudo: /var/lib/zentyal/tmp/ME0GadHCfP.cmd 2> /var/lib/zentyal/tmp/stderr', 'systemctl start \'samba-ad-dc\'', 256, 'ARRAY(0x55f3d2f849c0)', 'ARRAY(0x55f3ce00cfc8)') called at /usr/share/perl5/EBox/Sudo.pm line 210
EBox::Sudo::_root(1, 'systemctl start \'samba-ad-dc\'') called at /usr/share/perl5/EBox/Sudo.pm line 153
EBox::Sudo::root('systemctl start \'samba-ad-dc\'') called at /usr/share/perl5/EBox/Service.pm line 49
EBox::Service::manage('samba-ad-dc', 'start') called at /usr/share/perl5/EBox/Module/Service.pm line 746
EBox::Module::Service::_manageDaemon('EBox::Samba=HASH(0x55f3d14e9290)', 'HASH(0x55f3d2f84690)', 'start') called at /usr/share/perl5/EBox/Module/Service.pm line 728
EBox::Module::Service::_startDaemon('EBox::Samba=HASH(0x55f3d14e9290)', 'HASH(0x55f3d2f84690)') called at /usr/share/perl5/EBox/Samba.pm line 509
EBox::Samba::_startDaemon('EBox::Samba=HASH(0x55f3d14e9290)', 'HASH(0x55f3d2f84690)') called at /usr/share/perl5/EBox/Module/Service.pm line 772
EBox::Module::Service::_manageService('EBox::Samba=HASH(0x55f3d14e9290)', 'start') called at /usr/share/perl5/EBox/Module/Service.pm line 797
EBox::Module::Service::_startService('EBox::Samba=HASH(0x55f3d14e9290)') called at /usr/share/perl5/EBox/Samba.pm line 937
EBox::Samba::_startService('EBox::Samba=HASH(0x55f3d14e9290)') called at /usr/share/perl5/EBox/Module/Service.pm line 996
EBox::Module::Service::_enforceServiceState('EBox::Samba=HASH(0x55f3d14e9290)') called at /usr/share/perl5/EBox/Module/Service.pm line 947
EBox::Module::Service::_regenConfig('EBox::Samba=HASH(0x55f3d14e9290)') called at /usr/share/perl5/EBox/Samba.pm line 670
EBox::Samba::_regenConfig('EBox::Samba=HASH(0x55f3d14e9290)') called at /usr/share/perl5/EBox/Module/Base.pm line 234
eval {...} at /usr/share/perl5/EBox/Module/Base.pm line 233
EBox::Module::Base::save('EBox::Samba=HASH(0x55f3d14e9290)') called at /usr/share/perl5/EBox/GlobalImpl.pm line 648
eval {...} at /usr/share/perl5/EBox/GlobalImpl.pm line 647
EBox::GlobalImpl::saveAllModules('EBox::GlobalImpl=HASH(0x55f3cc93efe0)', 'progress', 'EBox::ProgressIndicator=HASH(0x55f3cbf01d88)') called at /usr/share/perl5/EBox/Global.pm line 95
EBox::Global::AUTOLOAD('EBox::Global=HASH(0x55f3cfff4a88)', 'progress', 'EBox::ProgressIndicator=HASH(0x55f3cbf01d88)') called at /usr/share/zentyal/global-action line 32
eval {...} at /usr/share/zentyal/global-action line 30
2018/12/10 09:03:25 ERROR> GlobalImpl.pm:652 EBox::GlobalImpl::saveAllModules - Failed to save changes in module samba: root command systemctl start 'samba-ad-dc' failed. 
Error output: Job for samba-ad-dc.service failed because a timeout was exceeded.
 See "systemctl status samba-ad-dc.service" and "journalctl -xe" for details.

Command output: . 
Exit value: 1
2018/12/10 09:03:25 INFO> Service.pm:965 EBox::Module::Service::restartService - Restarting service for module: samba
2018/12/10 09:04:46 INFO> Base.pm:231 EBox::Module::Base::save - Restarting service for module: logs
2018/12/10 09:04:47 ERROR> GlobalImpl.pm:727 EBox::GlobalImpl::saveAllModules - The following modules failed while saving their changes, their state is unknown: samba  at The following modules failed while saving their changes, their state is unknown: samba  at /usr/share/perl5/EBox/GlobalImpl.pm line 727
EBox::GlobalImpl::saveAllModules('EBox::GlobalImpl=HASH(0x55f3cc93efe0)', 'progress', 'EBox::ProgressIndicator=HASH(0x55f3cbf01d88)') called at /usr/share/perl5/EBox/Global.pm line 95
EBox::Global::AUTOLOAD('EBox::Global=HASH(0x55f3cfff4a88)', 'progress', 'EBox::ProgressIndicator=HASH(0x55f3cbf01d88)') called at /usr/share/zentyal/global-action line 32
eval {...} at /usr/share/zentyal/global-action line 30

**Comment by lostboy1** [comment]
I was able to get around this by using the following commands:
```
service smdb stop
service nmbd stop
service winbind stop
dpkg --configure -a
```

**Comment by andrzejk2** [comment]
I have this same problem, any solutions?

`Konfigurowanie pakietu samba (2:4.7.6+dfsg~ubuntu-0ubuntu2.5) ...
initctl: Unable to connect to Upstart: Failed to connect to socket /com/ubuntu/upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel(s) for script `lxdm'
insserv: Default-Stop  undefined, assuming empty stop  runlevel(s) for script `lxdm'
initctl: Unable to connect to Upstart: Failed to connect to socket /com/ubuntu/upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel(s) for script `lxdm'
insserv: Default-Stop  undefined, assuming empty stop  runlevel(s) for script `lxdm'
initctl: Unable to connect to Upstart: Failed to connect to socket /com/ubuntu/upstart: Connection refused
insserv: warning: script 'lxdm' missing LSB tags and overrides
insserv: Default-Start undefined, assuming empty start runlevel(s) for script `lxdm'
insserv: Default-Stop  undefined, assuming empty stop  runlevel(s) for script `lxdm'
Job for samba-ad-dc.service failed because a timeout was exceeded.
See "systemctl status samba-ad-dc.service" and "journalctl -xe" for details.
invoke-rc.d: initscript samba-ad-dc, action "restart" failed.
● samba-ad-dc.service - Samba AD Daemon
   Loaded: loaded (/lib/systemd/system/samba-ad-dc.service; enabled; vendor preset: enabled)
   Active: failed (Result: timeout) since Mon 2018-12-31 16:53:07 CET; 8ms ago
     Docs: man:samba(8)
           man:samba(7)
           man:smb.conf(5)
  Process: 11732 ExecStart=/usr/sbin/samba --foreground --no-process-group $SAMBAOPTIONS (code=exited, status=127)
 Main PID: 11732 (code=exited, status=127)

gru 31 16:53:07 gim2 samba[11745]:   Exiting pid 11745 on SIGTERM
gru 31 16:53:07 gim2 samba[11746]: [2018/12/31 16:53:07.903555,  0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
gru 31 16:53:07 gim2 samba[11746]:   Exiting pid 11746 on SIGTERM
gru 31 16:53:07 gim2 samba[11744]: [2018/12/31 16:53:07.903666,  0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
gru 31 16:53:07 gim2 samba[11744]:   Exiting pid 11744 on SIGTERM
gru 31 16:53:07 gim2 samba[11742]: [2018/12/31 16:53:07.921086,  0] ../source4/smbd/process_standard.c:86(sigterm_signal_handler)
gru 31 16:53:07 gim2 samba[11742]:   Exiting pid 11742 on SIGTERM
gru 31 16:53:07 gim2 systemd[1]: samba-ad-dc.service: Main process exited, code=exited, status=127/n/a
gru 31 16:53:07 gim2 systemd[1]: samba-ad-dc.service: Failed with result 'timeout'.
gru 31 16:53:07 gim2 systemd[1]: Failed to start Samba AD Daemon.
dpkg: błąd przetwarzania pakietu samba (--configure):
 installed samba package post-installation script subprocess returned error exit status 1
dpkg: problemy z zależnościami uniemożliwiają skonfigurowanie pakietu zentyal-samba:
 zentyal-samba zależy od samba; jednakże:
  Pakiet samba nie jest jeszcze skonfigurowany.

dpkg: błąd przetwarzania pakietu zentyal-samba (--configure):
 problemy z zależnościami - pozostawianie nieskonfigurowanego
dpkg: problemy z zależnościami uniemożliwiają skonfigurowanie pakietu zentyal-mail:
 zentyal-mail zależy od zentyal-samba; jednakże:
  Pakiet zentyal-samba nie jest jeszcze skonfigurowany.

dpkg: błąd przetwarzania pakietu zentyal-mail (--configure):
 problemy z zależnościami - pozostawianie nieskonfigurowanego
dpkg: problemy z zależnościami uniemożliwiają skonfigurowanie pakietu zentyal-mailfilter:
 zentyal-mailfilter zależy od zentyal-mail; jednakże:
  Pakiet zentyal-mail nie jest jeszcze skonfigurowany.

dpkg: błąd przetwarzania pakietu zentyal-mailfilter (--configure):
 problemy z zależnościami - pozostawianie nieskonfigurowanego
dpkg: problemy z zależnościami uniemożliwiają skonfigurowanie pakietu zentyal-sogo:
 zentyal-sogo zależy od zentyal-mail; jednakże:
  Pakiet zentyal-mail nie jest jeszcze skonfigurowany.

dpkg: błąd przetwarzania pakietu zentyal-sogo (--configure):
 problemy z zależnościami - pozostawianie nieskonfigurowanego
Przetwarzanie wyzwalaczy pakietu libc-bin (2.27-3ubuntu1)...
Wystąpiły błędy podczas przetwarzania:
 samba
 zentyal-samba
 zentyal-mail
 zentyal-mailfilter
 zentyal-sogo
`

**Comment by brunolorente** [comment]
Thank you for reporting this. Please note that currently it is not technically feasible to provide a solution to this because we can't reproduce the issue. 

Closing the issue.

**Comment by Neustradamus** [comment]
@neovs @loxK @gadgetusaf @phaidros7 @lostboy1 @citystar-at: Have you found a solution?

**Comment by therabreal** [comment]
Thanks lostboy1 - solved my issue after upgrading to 6.1
